### PR TITLE
Add test for TypeTranslator in Hive connector

### PR DESCRIPTION
**Pull request by geraint0923**
Also update the integration test to fit TypeTranslator.

### CODE REVIEW

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java [hunk 1]
```diff
@@ -1174,8 +1182,20 @@ private boolean insertOperationsSupported(HiveStorageFormat storageFormat)
         return storageFormat != HiveStorageFormat.DWRF;
     }
 
+    private Type getMatchedType(Type type)
```
Comment (haozhun) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java]: I had a quick discussion with @electrum. Let's call this method `canonicalizeType`.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java [hunk 2]
```diff
@@ -1174,8 +1182,20 @@ private boolean insertOperationsSupported(HiveStorageFormat storageFormat)
...
+    {
+        HiveType hiveType = HiveType.toHiveType(typeTranslator, type);
+        return typeManager.getType(hiveType.getTypeSignature());
+    }
+
+    private String getMatchedTypeName(String type)
```
Comment (haozhun) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java]: Let's call this method `canonicalizeTypeName`.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveTypeTranslator.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    {
+        this.typeManager = new TypeRegistry();
+        this.typeTranslator = requireNonNull(typeTranslator, "typeTranslator is null");
+
+        ImmutableMap<String, HiveType> hiveTypeTranslationMap = ImmutableMap.<String, HiveType>builder()
+                .put("bigint", HiveType.HIVE_LONG)
+                .put("integer", HiveType.HIVE_INT)
+                .put("smallint", HiveType.HIVE_SHORT)
+                .put("tinyint", HiveType.HIVE_BYTE)
+                .put("varchar(3)", HiveType.valueOf("varchar(3)"))
```
Comment (haozhun) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveTypeTranslator.java]: add `double` and `varbinary`

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveTypeTranslator.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                .put("varchar", HiveType.HIVE_STRING)
+                .put("date", HiveType.HIVE_DATE)
```
Comment (haozhun) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveTypeTranslator.java]: add `timestamp` and `decimal`

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveTypeTranslator.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+                .put("array(timestamp)", HiveType.valueOf("array<timestamp>"))
+                .put("map(boolean, varbinary)", HiveType.valueOf("map<boolean,binary>"))
```
Comment (haozhun) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveTypeTranslator.java]: add `row`

**Comment by haozhun** [comment]
Looks good assuming all tests pass.

**Comment by geraint0923** [comment]
Addressed the comments, will merge after the tests pass.

**Pull request merged**